src/RequestLogger/Program.cs

```diff
@@ -28,25 +43,80 @@
     app.UseSwaggerUI();
 }
 
+Log.Information("Request logger settings: {@Settings}", app.Configuration.GetSection("RequestLoggerSettings").Get<RequestLoggerSettings>());
+
+RequestLoggerContextConfiguration.TryRunMigrations(builder.Configuration);
+
 app.UseHttpsRedirection();
 
 app.UseAuthorization();
 
-app.Use(async (context, next) =>
+app.Run(async (context) =>
 {
     context.Request.EnableBuffering();
```

Comment:
There's a few different things happening in this method - woult it read easier if split into a series of methods calls, even if some are one-liners. e.g.

```cs
app.Run(async (context) =>
{
    var requestBody = await GetJsonRequestBody(context);
    var customerId = TryGetCustomerId(context.Request);
    var queryStringDictionary = BuildQueryStringDict(context.Request);
    var service = TryGetServiceName(context.Request);
    
    var request = new Request
    {
      // set props...
    };
    
    var requestCompleted = await WriteLogMessage(request);

    await context.Response.WriteAsync(JsonSerializer.Serialize(requestCompleted));
});
```